msmbuilder/cluster/agglomerative.py

```diff
@@ -30,21 +27,29 @@
 
 __all__ = ['_LandmarkAgglomerative']
 
+
+def ward_pooling_function(x, cluster_cardinality, intra_cluster_sum):
+    normalization_factor = cluster_cardinality*(cluster_cardinality+1)/2
```

Comment:
Be careful with this - in python 2, it will do integer division (i.e. dividing by 2 vs 2.0 returns an integer and a float, respectively).